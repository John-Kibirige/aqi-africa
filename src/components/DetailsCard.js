import React from 'react';

const DetailsCard = (props) => {
  const {
    imageUrl,
    country,
    aqIndex,
    latitude,
    longitude,
    capital,
    
    co,
    no,
    no2,
    o3,
    so2,
    pm25,
    pm10,
    nh3,
  } = props;
  return (
    <div className="details-card">
      <img src={imageUrl} alt="" className="card-image" />
      <div className="details-text">
        <h1 className="card-title">{country}</h1>
        <h3 className="card-capital">{capital}</h3>
        <p className="aqi-value">{aqIndex}</p>
        <p className="lon-lat">
          Longitude: {longitude}, Latitude: {latitude}
        </p>
      </div>

      <table class="table-fixed">
        <thead>
          <tr>
            <th>Symbol</th>
            <th>Component name</th>
            <th>
              Concentration (Î¼g/m<sup>3</sup>)
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>CO</td>
            <td>Carbon Monoxide</td>
            <td>{co}</td>
          </tr>
          <tr>
            <td>NO</td>
            <td>Nitrogen monoxide</td>
            <td>{no}</td>
          </tr>
          <tr>
            <td>
              NO<sub>2</sub>
            </td>
            <td>Nitrogen dioxide</td>
            <td>{no2}</td>
          </tr>
          <tr>
            <td>
              O<sub>3</sub>
            </td>
            <td>Ozone</td>
            <td>{o3}</td>
          </tr>
          <tr>
            <td>
              O<sub>3</sub>
            </td>
            <td>Ozone</td>
            <td>{o3}</td>
          </tr>
          <tr>
            <td>
              SO<sub>2</sub>
            </td>
            <td>Sulphur dioxide</td>
            <td>{so2}</td>
          </tr>
          <tr>
            <td>
              PM<sub>2_5</sub>
            </td>
            <td>Small size particulate</td>
            <td>{pm25}</td>
          </tr>
          <tr>
            <td>
              PM<sub>10</sub>
            </td>
            <td>Bigger size Particulates </td>
            <td>{pm10}</td>
          </tr>
          <tr>
            <td>
              NH<sub>3</sub>
            </td>
            <td>Ammonia</td>
            <td>{nh3}</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

export default DetailsCard;
