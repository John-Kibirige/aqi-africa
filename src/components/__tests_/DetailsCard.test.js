import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DetailsCard from '../DetailsCard';

const renderedComp = (
  <DetailsCard
    imageUrl="www"
    country="String"
    aqIndex={43}
    latitude={43}
    longitude={43}
    capital="String"
    co={43}
    no={43}
    no2={43}
    o3={43}
    so2={43}
    pm25={43}
    pm10={43}
    nh3={43}
  />
);

it('Render details card successfully', () => {
  render(renderedComp);
  const element = screen.getByText(/Air Quality Index breakdown/i);
  expect(element).toBeInTheDocument();
});

it('Create snapshot test for the Details Card component ', () => {
  const tree = renderer.create(renderedComp).toJSON();
  expect(tree).toMatchSnapshot();
});
