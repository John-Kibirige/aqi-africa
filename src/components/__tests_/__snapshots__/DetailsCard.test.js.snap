// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create snapshot test for the Details Card component  1`] = `
<div
  className="details-card min-h-screen h-full details-container p-4 bg-slate-200 sm:px-7 md:pt-8"
>
  <div
    className="img-text relative flex sm:flex sm:gap-4"
  >
    <img
      alt="String"
      className="card-image h-[28vh] object-contain w-[45%] sm:h-[36vh] sm:object-contain md:w-2/5"
      src="www"
    />
    <div
      className="details-text  shadow-xl p-3 bg-white ml-3 w-[50%]  top-3 right-2 $sm:static md:w-3/12"
    >
      <h1
        className="card-title uppercase text-pink-500 font-bold text-center sm:text-2xl sm:mt-2"
      >
        String
      </h1>
      <h3
        className="card-capital text-center underline sm:text-xl sm:no-underline sm:font-bold sm:my-2"
      >
        String
      </h3>
      <p
        className="lon-lat sm:text-xl sm:text-center sm:mb-2"
      >
        Longitude: 
        43
      </p>
      <p
        className="lon-lat sm:text-xl sm:text-center sm:mb-2"
      >
        Latitude: 
        43
      </p>
    </div>
    <div
      className="rubric-table ml-auto mr-14 hidden md:block mt-10"
    >
      <table
        className="table-fixed"
      >
        <thead>
          <tr>
            <th
              className="p-1"
            >
              Rubric
            </th>
            <th
              className="p-1"
            >
              Index
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              className="p-1 font-normal border-b border-pink-400 border-r"
            >
              Good
            </th>
            <th
              className="p-1 font-normal border-b border-pink-400 "
            >
              1
            </th>
          </tr>
          <tr>
            <th
              className="p-1 font-normal border-b border-pink-400 border-r"
            >
              Fair
            </th>
            <th
              className="p-1 font-normal border-b border-pink-400 "
            >
              2
            </th>
          </tr>
          <tr>
            <th
              className="p-1 font-normal border-b border-pink-400 border-r"
            >
              Moderate
            </th>
            <th
              className="p-1 font-normal border-b border-pink-400 "
            >
              3
            </th>
          </tr>
          <tr>
            <th
              className="p-1 font-normal border-b border-pink-400 border-r"
            >
              Poor
            </th>
            <th
              className="p-1 font-normal border-b border-pink-400 "
            >
              4
            </th>
          </tr>
          <tr>
            <th
              className="p-1 font-normal  border-pink-400 border-r"
            >
              Very Poor
            </th>
            <th
              className="p-1 font-normal "
            >
              5
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <button
    className="index bg-pink-500 text-gray-100 uppercase px-3 sm:px-5 py-1 sm:py-2 sm:text-white sm:font-bold sm:text-lg rounded ml-auto hover:bg-pink-400 block mt-[18px] mr-3 sm:mt-7 sm:mr-7"
    type="button"
  >
    index = 
    43
  </button>
  <h2
    className="break-down font-bold text-2xl uppercase text-center my-4 sm:text-3xl"
  >
    Air Quality Index breakdown
  </h2>
  <table
    className="table-fixed sm:w-full mb-6"
  >
    <thead>
      <tr
        className="bg-pink-600 text-white odd:border border border-solid"
      >
        <th
          className="header w-1/3 sm:p-3 sm:text-lg"
        >
          Symbol
        </th>
        <th
          className="header w-1/3 sm:p-3 sm:text-lg"
        >
          Component name
        </th>
        <th
          className="header w-1/3 sm:p-3 sm:text-lg"
        >
          Concentration (Î¼g/m
          <sup>
            3
          </sup>
          )
        </th>
      </tr>
    </thead>
    <tbody
      className="text-center"
    >
      <tr>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          CO
        </td>
        <td
          className="hover:bg-pink-200 transition-all border border-solid border-pink-400"
        >
          Carbon Monoxide
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-l border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          NO
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          Nitrogen monoxide
        </td>
        <td
          className="border border-solid border-r border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          NO
          <sub>
            2
          </sub>
        </td>
        <td>
          Nitrogen dioxide
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-l border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          O
          <sub>
            3
          </sub>
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          Ozone
        </td>
        <td
          className="border border-solid border-r border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-l border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          SO
          <sub>
            2
          </sub>
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
           
          Sulphur dioxide
        </td>
        <td
          className="border border-solid border-r border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          PM
          <sub>
            2_5
          </sub>
        </td>
        <td
          className="hover:bg-pink-200 transition-all"
        >
          Small size particulates
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-l border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          PM
          <sub>
            10
          </sub>
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
           
          Bigger size Particulates
        </td>
        <td
          className="border border-solid border-r border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
      <tr>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all sm:p-2"
        >
          NH
          <sub>
            3
          </sub>
        </td>
        <td
          className="border border-solid border-b border-pink-400 hover:bg-pink-200 transition-all"
        >
          Ammonia
        </td>
        <td
          className="border border-solid border-pink-400 hover:bg-pink-200 transition-all"
        >
          43
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
